### PR TITLE
feat: multi host decorator

### PR DESCRIPTION
**Pull request by elrrrrrrr**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)

controller-decorator

##### Description of change

Make the `@Host` decorator supports multiple host

@Host(string) => @Host(string | string[])

### CODE REVIEW

File: core/controller-decorator/src/decorator/http/Host.ts [hunk 1]
```diff
@@ -2,13 +2,14 @@ import ControllerInfoUtil from '../../util/ControllerInfoUtil';
 import { EggProtoImplClass } from '@eggjs/core-decorator';
 import MethodInfoUtil from '../../util/MethodInfoUtil';
 import assert from 'assert';
+import { HostType } from '../../model';
 
-export function Host(host: string) {
+export function Host(host: HostType) {
```
Comment (killagu) [inline core/controller-decorator/src/decorator/http/Host.ts:7]: string | string[] 应该只在这里使用。下层模型统一用 string[]

Comment (elrrrrrrr) [reply core/controller-decorator/src/decorator/http/Host.ts:7]: 好的 那后续 meta 中的 host 就保持 string [] | undefined 了 😈

Comment (elrrrrrrr) [reply core/controller-decorator/src/decorator/http/Host.ts:7]: > string | string[] 应该只在这里使用。下层模型统一用 string[]

好的 那后续 meta 中的 host 就保持 string [] | undefined 了 😈

File: plugin/controller/lib/impl/http/HTTPMethodRegister.ts [hunk 2]
```diff
@@ -49,15 +49,22 @@ export class HTTPMethodRegister {
     this.eggContainerFactory = eggContainerFactory;
   }
 
-  private createHandler(methodMeta: HTTPMethodMeta, host: string | undefined) {
+  private hostMatch(host: string[], target: string) {
+    if (Array.isArray(host)) {
```
Comment (fengmk2) [inline plugin/controller/lib/impl/http/HTTPMethodRegister.ts:53]: 参数明明声明 host 是 array，为啥还要做一次判断？

**Review by killagu — APPROVED** [review]
LGTM

**Comment by killagu** [comment]
- @eggjs/controller-decorator@1.7.0
 - @eggjs/tegg-aop-plugin@1.5.5
 - @eggjs/tegg-controller-plugin@1.7.5
 - @eggjs/tegg-eventbus-plugin@1.5.5
 - @eggjs/tegg-orm-plugin@2.4.5
 - @eggjs/tegg-plugin@1.5.5
 - @eggjs/tegg-schedule-plugin@2.4.5
 - @eggjs/tegg-standalone@1.5.4
 - @eggjs/tegg@1.6.4

**Pull request merged**